src/beliefs/discrete.jl

```diff
@@ -1,133 +1,107 @@
-# Maintained by Max Egorov and Zach Sunberg
+# Goals: minimize calls to ordered_states (allocates memory)
 
-"""
-A weight vector to be used as a belief or state distribution.
-
-sum(b)=1 is not enforced at all times, but during the update it is normalized to sum to one.
-
-pdf(b, i) calculates the sum every time it is called. To access the weight directly (for example if you are sure that the sum is 1), use weight(b, i). 
-"""
-mutable struct DiscreteBelief
+# needs pomdp for state_index in pdf(b, s)
+# needs list of ordered_states for rand(b)
+mutable struct DiscreteBelief{P<:POMDP, S}
```

Comment:
Does it need to be mutable? I think it can just be a struct.

Reply:
I think you are correct. However, every time we create a new belief from an old belief, we pass in the old belief's state list. We've said this won't be a big deal because it is just copying the reference. But if the belief is immutable, will the state list be copied over instead? I don't think so but you probably know better so I'll leave it up to you.

Reply:
If you have an immutable struct with a field that is an array, you can still mutate the array. You just can't point that field to a different array or something else. You would still be copying the reference, not the contents of the array.

Reply:
Cool, thanks for letting me know.